docs/proposals/logdir.md

```diff
@@ -0,0 +1,163 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/logdir.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# LogDir
+
+## Abstract
+
+A proposal for implementing `LogDir`. A `LogDir` can preserve and manage a pod's log files.
+
+Several existing issues and PRs were already created regarding that particular subject:
+* Collect logfiles inside the container [#12567](https://github.com/kubernetes/kubernetes/issues/12567)
+* Add LogDir [#13010](https://github.com/kubernetes/kubernetes/pull/13010#issuecomment-133997188)
+
+Author: WuLonghui (@wulonghui)
+
+## Motivation
+
+Some applications will write log to files, so need to add  `LogDir`. It can add some useful infomation to make logging agent easy to collect the logs, also we can manage the log files(like log rotation and maximum log file sizes).
+
+## Implementation
+
+Create `LogDir` as a policy  to `api.VolumeMount` struct :
+
+```
+// VolumeMount describes a mounting of a Volume within a container.
+type VolumeMount struct {
+  // Required: This must match the Name of a Volume [above].
+  Name string `json:"name"`
+  // Optional: Defaults to false (read-write).
+  ReadOnly bool `json:"readOnly,omitempty"`
+  // Required.
+  MountPath string `json:"mountPath"`
+  // Optional: Policies of VolumeMount.
+  Policy *VolumeMountPolicy `json:"policy,omitempty"`
+}
+
+// VolumeMountPolicy describes policies of a VolumeMount.
+type VolumeMountPolicy struct {
+  // Optional: LogDir policy.
+  LogDir *LogDirPolicy `json:"logDir,omitempty"`
+}
+
+// LogDirPolicy describes a policy of logDir, include log rotation and maximum log file size.
+type LogDirPolicy struct {
+  // Optional: Glob pattern of log files.
+  Glob string `json:"glob,omitempty"`

```

Comment:
Who is going to consume the parameters and how?  Are these consumed by fluentd (or whatever) or by a kube component?  Are they generic or particular to a given logging consumer?